internal/preferences.go

```diff
@@ -11,17 +11,22 @@ type InMemoryPreferences struct {
 	values          map[string]interface{}
 	lock            sync.RWMutex
 	changeListeners []func()
+	waitGroup       *sync.WaitGroup
 }
 
 // Declare conformity with Preferences interface
 var _ fyne.Preferences = (*InMemoryPreferences)(nil)
 
 // AddChangeListener allows code to be notified when some preferences change. This will fire on any update.
+// listener should not try to write values
 func (p *InMemoryPreferences) AddChangeListener(listener func()) {
 	p.lock.Lock()
 	defer p.lock.Unlock()
 
-	p.changeListeners = append(p.changeListeners, listener)
+	p.changeListeners = append(p.changeListeners, func() {
+		defer p.waitGroup.Done()
```

Comment:
Can't we just place it after the other function instead of deferring?

Reply:
No, it's safer that way
a deferred function will always be evaluated now and executed later, while if I placed it after the call to `listener()` it may not get executed if the called function raised a panic

Reply:
Ah, I see. Thanks 👍